cmd/limactl/edit.go

```diff
@@ -53,14 +58,30 @@ func editAction(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	hdr := fmt.Sprintf("# Please edit the following configuration for Lima instance %q\n", instName)
-	hdr += "# and an empty file will abort the edit.\n"
-	hdr += "\n"
-	hdr += editutil.GenerateEditorWarningHeader()
-	yBytes, err := editutil.OpenEditor(instName, yContent, hdr)
+	tty, err := cmd.Flags().GetBool("tty")
 	if err != nil {
 		return err
 	}
+	yq, err := cmd.Flags().GetString("set")
+	if err != nil {
+		return err
+	}
+	var yBytes []byte
+	if yq != "" {
+		yBytes, err = yqutil.EvaluateExpression(yq, yContent)
+		if err != nil {
+			return err
+		}
+	} else {
+		hdr := fmt.Sprintf("# Please edit the following configuration for Lima instance %q\n", instName)
+		hdr += "# and an empty file will abort the edit.\n"
+		hdr += "\n"
+		hdr += editutil.GenerateEditorWarningHeader()
+		yBytes, err = editutil.OpenEditor(instName, yContent, hdr)
+		if err != nil {
+			return err
+		}
```

Comment:
Needs to check `tty` before opening an editor

Reply:
Not necessarily, that would depend on the editor ?

Reply:
Unfortunately you can't give a command (like yq) as EDITOR, but you do get an error message if it doesn't work

```
$ _output/bin/limactl --debug edit default </dev/null
DEBU[0000] opening editor "/usr/bin/editor" for a file "/tmp/lima-editor-996832437" 
Too many errors from stdin
FATA[0000] could not execute editor "/usr/bin/editor" for a file "/tmp/lima-editor-996832437": exit status 1 
```

Reply:
When `--tty` is true but the tty is not available, it should fail with a human-readable error

Reply:
@afbjorklund WDYT?

Reply:
I will take a look at what `start` is doing, it has the same strange connection between tty and editor

```
      --tty                enable TUI interactions such as opening an editor, defaults to true when stdout is a terminal (default true)
```

Reply:
Now it will only open the editor if a tty is present, but will not ask to start without a tty.

The user can run `limactl start` after `limactl edit` explicitly, if the tty is not available.